packages/api/src/experimental/subscribe.ts

```diff
@@ -22,21 +22,23 @@ import type {
 import type { EXPERIMENTAL_ObjectSetListener } from "../public/unstable.js";
 import type { Experiment } from "./Experiment.js";
 
+type NonEmptyArray<T> = [T, ...T[]];
```

Comment:
I'm 80% sure we shouldn't use this type. I've seen this type show up in other code bases where it just ends up proliferating all over the place causing more harm than good.

Reply:
I see, how should we enforce a non empty array here then? Does it make sense to throw if we receive an empty array? I don't want to accept empty arrays because that will then request all non reference backed properties and some users might become reliant on that. Also curious to know what issues it creates

Reply:
Discussed offline, will add a throw here on a FLUP